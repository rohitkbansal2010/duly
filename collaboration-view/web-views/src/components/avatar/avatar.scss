@import "~@styles/colors.scss";
@import "~@styles/typography.scss";

.avatar {
  display: flex !important;
  flex-shrink: 0 !important;
  border-radius: 50%;

  img {
    z-index: 2;
    width: 100%;
    height: 100%;
  }

  &Fallback {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 1 !important;
  }

  &Lettered {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    display: flex !important;
    align-items: center !important;
    justify-content: center !important;
    width: 100% !important;
    height: 100% !important;
    color: $white;
    font-weight: 600;
    line-height: 100%;
    border-radius: 50%;
    @include nexa-extra-bold-highlight;

    &MedicalDoctor,
    &RegularEmployee {
      margin: 0 auto !important;
    }

    &MedicalDoctor {
      background-color: $dupage-dark-blue;
    }

    &RegularEmployee {
      background-color: $dupage-magenta;
    }

    &Transparent {
      background-color: transparent;
    }
  }
}

.borderAvatar {
  border-color: $white;
  border-style: solid inset solid solid;

  &RegularEmployee,
  &MedicalDoctor,
  &NonEmployee {
    @extend .borderAvatar;
    outline-style: solid;
  }

  &NonEmployee {
    outline-color: $white;
    box-shadow: 0 0.125rem 0.25rem -0.0625rem rgb(102 110 125 / 6%),
      0 0.25rem 0.375rem -0.0625rem rgb(102 110 125 / 10%), 0 0 0.0625rem 0 rgb(102 110 125 / 6%);
  }

  &RegularEmployee {
    outline-color: $dupage-violet;
  }

  &MedicalDoctor {
    outline-color: $dupage-light-blue;
  }
}
