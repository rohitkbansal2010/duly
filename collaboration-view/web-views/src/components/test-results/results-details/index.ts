export { ResultsDetails } from './results-details';
