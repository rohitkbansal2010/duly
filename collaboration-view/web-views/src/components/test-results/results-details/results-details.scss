@import "~@styles/typography";
@import "~@styles/colors";

@mixin ordered {
  width: 25%;
}

@mixin collected {
  width: 30%;
}

@mixin resulted {
  width: 30%;
}

@mixin status {
  width: 15%;
}

.results {
  &Details {
    display: flex;
    flex-direction: column;
    margin-top: 1.25rem;
    padding: 1.25rem;
    background: $light-gray;
    border-radius: 0.5rem;

    &Header {
      @include inter-text-xs-bold;
      display: flex;

      color: $dark-gray;
      font-size: var(--font-size-16px-4px);
      text-transform: uppercase;

      &Ordered {
        @include ordered;
      }

      &Collected {
        @include collected;
      }

      &Resulted {
        @include resulted;
      }

      &Status {
        @include status;

        img {
          width: 0.7rem;
          height: 0.7rem;
          margin-left: 0.3rem;
        }
      }
    }

    &Body {
      @include inter-text-default-bold;
      display: flex;
      font-size: var(--font-size-14px-4px);

      &Ordered {
        @include ordered;

        display: flex;
        align-items: center;

        &Doctor {
          margin-left: 0.5rem;
        }
      }

      &Collected {
        @include collected;

        display: flex;
        align-items: center;
      }

      &Resulted {
        @include resulted;

        display: flex;
        align-items: center;
      }

      &Status {
        @include status;

        display: flex;
        align-items: center;

        &Indicator {
          width: 0.75rem;
          height: 0.75rem;
          margin-left: 0.5rem;
          background-color: $dupage-saffron;
          border-radius: 50%;

          &Final {
            background-color: $dupage-green;
          }

          &Cancelled,
          &Unknown {
            background-color: $dupage-orange;
          }
        }
      }
    }
  }
}
