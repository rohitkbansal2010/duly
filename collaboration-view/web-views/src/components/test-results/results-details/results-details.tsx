import { Avatar } from '@components/avatar';
import { TestResultStatus } from '@enums';
import { Performer } from '@types';
import {
  getSrcAvatar,
  getTestReportDate,
  getUserRole,
  showFirstPrefixItem,
  getTestReportTime,
  getPerformerInfo
} from '@utils';

import styles from './results-details.scss';

type ResultsDetailsProps = {
  effectiveDate?: string;
  issued?: string;
  status?: TestResultStatus;
  performers?: Performer[];
}

export const ResultsDetails = ({
  effectiveDate = '', issued = '', status, performers = [], 
}: ResultsDetailsProps) => {
  const {
    familyName,
    givenNames,
    prefixes,
    photo,
    role,
  } = getPerformerInfo(performers);

  const indicatorClasses = []
    .concat(styles.resultsDetailsBodyStatusIndicator)
    .concat(styles[`resultsDetailsBodyStatusIndicator${status}`])
    .join(' ');
  
  return (
    <div className={styles.resultsDetails}>
      <div className={styles.resultsDetailsHeader}>
        <span className={styles.resultsDetailsHeaderOrdered}>ordered by</span>
        <span className={styles.resultsDetailsHeaderCollected}>collected on</span>
        <span className={styles.resultsDetailsHeaderResulted}>resulted on</span>
        <span className={styles.resultsDetailsHeaderStatus}>status</span>
      </div>
      <div className={styles.resultsDetailsBody}>
        <div className={styles.resultsDetailsBodyOrdered}>
          {performers[0] && (
            <Avatar
              width={2.125}
              src={getSrcAvatar(photo)}
              alt={`${showFirstPrefixItem(givenNames)}${familyName}`.trim()}
              role={getUserRole({ role, prefixes })}
            />
          )}
          <span className={styles.resultsDetailsBodyOrderedDoctor}>
            {prefixes && prefixes.join(' ')}
            {familyName}
          </span>
        </div>
        <span className={styles.resultsDetailsBodyCollected}>
          {effectiveDate &&
            `${getTestReportDate(effectiveDate)} - ${getTestReportTime(effectiveDate)}`}
        </span>
        <span className={styles.resultsDetailsBodyResulted}>
          {issued && `${getTestReportDate(issued)} - ${getTestReportTime(issued)}`}
        </span>
        <span className={styles.resultsDetailsBodyStatus}>
          {status}
          <div className={indicatorClasses} />
        </span>
      </div>
    </div>
  );
};
