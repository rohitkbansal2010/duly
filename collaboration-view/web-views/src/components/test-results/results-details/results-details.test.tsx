import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Avatar } from '@components/avatar';
import { testReportsResultMockData } from '@mock-data';
import {
  getPerformerInfo,
  getSrcAvatar,
  getTestReportDate,
  getTestReportTime,
  getUserRole,
  showFirstPrefixItem
} from '@utils';

import { ResultsDetails } from './results-details';

describe('ResultsDetails', () => {
  let component: ReturnType<typeof shallow>;

  const {
    effectiveDate,
    issued,
    status,
    performers,
  } = testReportsResultMockData;
  
  const mockProps = {
    effectiveDate,
    issued,
    status,
    performers,
  };
  
  const {
    familyName,
    givenNames,
    prefixes,
    photo,
    role,
  } = getPerformerInfo(performers);

  beforeEach(() => {
    jest
      .spyOn(Date.prototype, 'toLocaleTimeString')
      .mockImplementation(() =>
        '10:17 AM');

    component = shallow(<ResultsDetails {...mockProps} />);
  });

  it('component should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('component should contain right titles', () => {
    expect(component.text().includes('ordered by')).toBeTruthy();
    expect(component.text().includes('collected on')).toBeTruthy();
    expect(component.text().includes('resulted on')).toBeTruthy();
    expect(component.text().includes('status')).toBeTruthy();
  });

  it('component should contain avatar', () => {
    const avatar = component.find(Avatar);

    const expectedProps = {
      width: 2.125,
      src: getSrcAvatar(photo),
      alt: `${showFirstPrefixItem(givenNames)}${familyName}`.trim(),
      role: getUserRole({ role, prefixes }),
    };
    
    expect(avatar).toBeTruthy();
    expect(avatar.props()).toEqual(expectedProps);
  });

  it('component should contain right data', () => {
    const formattedEffectiveDate = `${getTestReportDate(effectiveDate)} - ${getTestReportTime(effectiveDate)}`;
    const formattedIssuedDate = `${getTestReportDate(issued!)} - ${getTestReportTime(issued!)}`;

    expect(component.text().includes(mockProps.status)).toBeTruthy();
    expect(component.text().includes(formattedEffectiveDate)).toBeTruthy();
    expect(component.text().includes(formattedIssuedDate)).toBeTruthy();
  });

  it('component should work with empty data', () => {
    component = shallow(<ResultsDetails />);

    expect(component.text().includes('Invalid Date')).toBeFalsy();
    expect(component.text().includes('undefined')).toBeFalsy();
    expect(toJson(component)).toMatchSnapshot();
  });
});
