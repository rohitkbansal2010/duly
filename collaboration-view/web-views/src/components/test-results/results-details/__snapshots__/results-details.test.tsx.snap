// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsDetails component should match snapshot 1`] = `
<div
  className="resultsDetails"
>
  <div
    className="resultsDetailsHeader"
  >
    <span
      className="resultsDetailsHeaderOrdered"
    >
      ordered by
    </span>
    <span
      className="resultsDetailsHeaderCollected"
    >
      collected on
    </span>
    <span
      className="resultsDetailsHeaderResulted"
    >
      resulted on
    </span>
    <span
      className="resultsDetailsHeaderStatus"
    >
      status
    </span>
  </div>
  <div
    className="resultsDetailsBody"
  >
    <div
      className="resultsDetailsBodyOrdered"
    >
      <Avatar
        alt="Ana Reyes"
        role="MedicalDoctor"
        src="data:image/x-png;base64,iVBORw0KGgoAAAANSUhEUgAAAEAAAABACAIAAAAlC+aJAAAABGdBTUEAALGPC/xhBQAAAAFzUkdCAK7OHOkAAAAgY0hSTQAAeiYAAICEAAD6AAAAgOgAAHUwAADqYAAAOpgAABdwnLpRPAAAHTJJREFUaIFdemlzpcd13lm6+13ugh2YDRjMDDkczpAiTS2kJEpy4lJix45TFVeSSiVViZ0PcariX5MvqVSqnJSjWLErsZ3IS2w5lhzLkm2JIilK4jobOIMBMADu+q7d5+RDvxczyvsJF8C993T3Oc/znOc0vvHmW6oqACKCiIgIAMRESMxMRN77umlC8CqqCiEIACaJJUJVjW9R1SBBRACAiKyxANC0jfeekIw1CAgACgAAtPiWEAIAqYKqMhMzhxCIMH4yPPXo4okviShGCwBGAQARAYgo/kgAoCAaVNVai4iECMRA8c0CoCKqCogAAEGCxsV1T7cwQlLVum2CBEQURWZmZkDgGBaoagBEYmImREABVVElAIjbAYvAzlYSvzVGDwBGsduZuJHx1wjdp8VPOTsZRCQiVQXAtvXx0xW66A2bswMRCaqBAI4O9u/fu5dm2WBpeWV1dTgcWrbChITeewFgZkOMCACKBKCIRN77EHyMhIiQEATivgAAEmrQ+NIgYgjh7G+wiPdsSWGxBhEJEgCAkBCRmb33AMDMqkpIAKIiIcje3r2Dw0f7Dx68/977P37/PRQNQdjYpeWl3d3dz7z62s2bN5eGS00QNExEgIhIqoKAgKAiKoKAoqKgIQRmDhJCEARAQgxPDgH/5vtvBhFVpZj9iERExAAgoIgYFlEGCd6HuNmIBKreB2JyziCiiqiER/uP/uD3f//7b35vPp81dSshICEBqQozJ2kGiMR847nnvvDFL9184QVjLRlj2XQJDKAAIgEUFDAEH0KIxw7a5WrcOyaOGYXfffMtUVHRuKLuHBCYOH5o27aimjiHiG3beu/jOagqKLDhmJvi2++98cZ/++pX57OxqpZlWRYNM6kqIjEzIgBq4pIsy5xzgvSLv/ALr3/pi9Y4tpaZEZCZRVQEiBAAvPchCGKXw10KEACAYSZAAOB/829/jYmMMYa7BxFV9OznuDHdDhEBaPBBQRGQjSEiVfC+/c63v/2V//IbZVE0VakimxubbVOH4IktAEJMZYAgIfhQ1zUhfnxvr56Vl7d3jLWLAoWn4AdVBRGI6El6UyxQJEQiJiJDiBpxKib6oh7O3sbMCiCqKgIAEgQQENAaG0JoGg8a3n7r7d/8zd+sylIlIGKSuJ3t874t9/fLpmlFlBmJEmuNtdYHb4xBhdlk8s4bbw56vS/+7JfjJpOSqoLiUzDT/QURRJ5gnahiCIhomrYFVREh7urpDKe7XQFgIlUVUAmxoJ/aEgl3Pnzvq1/5z6cnR8hIxBKCCzYnvry+Op/MZo16L8G3VVWIOgJkIhIgC1mWTaaj7373r7d2Lj1/8xYZg6IRXmPanB0FcoeMLKzSYaMAACjVVdU0DSAS4aKCuyfEIpIFRi0Ol4kIEVRBxLfNb//Wb+0/eAAS1AcjMkzszsZSH+vzS+lGbnusKUnqbJpmEqSuq8h0eZovLQ1tnowmoz//P39azKe0CHjBj2SMeWoZsAiOI1d1wfyLX/4VNuysjfSmoPTUBosIgAKiqMpTaBsPSYL/n7/3O9/+1v9t2tZamydZz5obOxdef+WFaxfXljO7sTxcHWbqKxDxCkhMgKDay/N+v28t9YeDeVlQEJul25cvM3HHuAi0wBlVDRo6Fupi0zN2MsR0VuPee0BQ0qeWrqqqEmLawUI7IIIG2X/44Bt/+nXSkLjEEthQPXfp4qsvPvPslQsJI2yuXd4OZVk+vHr+w9v37zyYfTyaeyQ2xlqTJo4QEmc319dPj44/eucHn/70a9y3igBAKhBQVBZ8rKCqokIUo1cijBrEGDbGGCJqvRdVEAV9qvAVVDoRQgs8i+vybfvNb/zZdDKyZMULQdg5v/TSjd0rF7YyS44NM/VzCv1sdal/befi0VHxxru337n9sCG2iVWVNMsJaXk4KGezk+OT+/fuXn/+FjIvtggBu6yNygAVF2UZQ6IINlH7KKiele+ZDokRi2qsBACIxO69Pzw6+KvvfNtYW3tRgIThxpWLl89tEAQNARFA1RBlWTocDi/vXH71sz/1T3/pF/7Wqy9vrAwsm8Q5JmBUw7SxsTEtinfffks1PCXinmT/2aMas6erh05KgIKInukcjEs6U5oAhKjAxhhVAQTxQSW88/bbxyfHDOAFCPwzly8+e/lCmhBCEAkhtGgsECASsU16A86Hl1bd3+E0Hwzf+OBu7dUa44NXlSxzVV0fPvh4Ph31l9cBGRGIUJRQ5UyqAZwJOUYkERGRLtbgfVwwIjIRMauqiMSImR2zJWI2RkTFtyL+O9/5jvfSeE+Ey7305tVLy70+k2E2hIxoEClKEzaWTc+YnFx2cWf79c986pUbzzL6eTGLnI1IAAheHn50mxEAhRiZ0VjjkoStRSYgRGYlEsUgGkTJGCQ2UbFpdzYQoTQyQ3cOXe5DCMEYRgBVnc/mt2/faZs2yxLxfnt97eLWZprkiIjAhMREjKQSAAwTIxEiAXLSz87vmBfb+ng+OhjPHh+fZGk2m81821Z1ub//4BaRgiEkAFIABSCmmFAq2uU6gKqABzpDzCDiQ4Qq6kKHDs6eJF9X9YbZnI5G48mk9cH70E/tzeu7eZYcnIxH87r2orJ4VDFqTAgq6lupgiT9/u7l7eevX00cJ2k+L8rj4+OmrgXCeHy6QOozDJQuDH1CrB22ShBV06mGEELwhJYQRUUViBgBRAUAAAVAkHShariu69ZL632f0mcvbmwNs2/99V/vHU0UzPmlwSefv3r54rl+3kvz3Hudnjw+/OC940ktysvrK7vXruzsXP6pF2/t3b8/nfv9/f3D/YM0y5u6KSaTupinfdfhCsRmEeAMf1SjGtK4Jaom6uS4JmZGIvUiGqyxiKhdixQ7IVrkGxhjVFWCNk3z3LXt8xsrG6ufOpwWICp1czKa+HL2zLVrw9WVAHpwZ29alEZheX3jwva5LEkOHh1sbq2/9plPfnj/D5p5PZnOHj8+vbZ9oazK2XiaD9YieyFjR68dkmvsd6Lqi5BjIuISEWonGDFy1k/CV6SwyMdN06Rp6pwti9IZs7t9rvXhf/zvb9XoDo+Pnzm/+cqz295XbdsMV5Zd3p89PpjO67c+uCt3Hn4SzRde311bWS2r4vL2pa3N9R99eG+j7/bLgomqupxOxusiogIIhMQcYeZJIEhkDACCAKgoBQlIlDjnrGWkmO0IKCKistj+7hEJse9Ok6SXOcNojQxzTqw+f3XrhasXXrl+ZXWpPxpPjLWcJpQN0/5KJfDBnb1JGY5H8+m8/v3/9Xt1WVimhHltdcml9ua1qytZrj5IXU9Gxz60GiWbqIgubAAEQI0ZjoBIjMiERlUJwVgbizZIh7sKID7E6onnd8ZuAmCZhnlyDNJPktzYlUH26gvXJ5NCzi0HCbHJdElGLlVjls9deOlW/UlrNi9e2Lh4UUXL+WwwXGKiPHOXN1ce3d1L02Q6mW5sLIemUQlkuAN9eKqkF2ygAIpKqgBqAFBUQwixoUGNxAyEcbmLMlqAQsyntmmNaMo2IUuiRvH85tbWJpdF3dR1UZa+rbMsTVzKNl1d3khvUNvU7LiePE6zfp6tMxGKXFoavnxx85sPDonNfD5fX1uuqiK23EigT3qCKCnxJ4FRNRoYACCiQURAmcg5xwuiRsQFnp69D5m49Y0SusTOy7JtJLReg3dICRODGkJmdMaxsWSsVdamtQQMmNi8N1xN8n5Us+eHg6wupK7SNKmbRsU3bUMASISAqAoiCNqBvWpkEwQEWXhEi6ggJruXoKo/0cUpIHUNmiogIKk2TV0F79JkPJsXTdsECW3bVGVVzmbTUVOWbNgQsYJWZWjaUFWhblA0sUma9pgtItRljb4at+3d/YNenqGqc9bXjQIECSKxERRUBYk4ChJEQhDvg18I7KeBpvVtUzfRLDmjDCTsALQ7S2Bjy6oSUZc4UT0tqiageIwOzWQyyXspIUpXMLxxabeu2/HjY/VBGckZUPBNW1VlurT8J29/MA2YWBdUEbSqCo0g1JWfLCJ5okPPWLI7gQgvvvVN3dRNU9f1Wb0ykWHDC8+DiAxzCL6p27ZtRMQ6dzKdCWJQYuNU0Rg7Pp08erDv66ZtqmoyCYqDQb+YTJuykNBI8MH7tm3nZfXrv/vHP7yzn/US4yhmaWgrFXmyudrt2k8yNJx1vRRdpDPRb5iRUBaFG8sg8kNU5V2mBa+iAGptMh6VCNQSnEwnx+MxuwyNG6YDrdrDe7cZsPZh/dqtyzdfmk6m7CVUbVlOBeTXv/Lbv/G7fwRAOfFSf8CW0ySBeo5SAYqqtm2IBoOCAAqgAAEQARExI7IqmjPRBnGDjYGF99ZV/eJBpFj40SBoW0+AAvLg8XELSr7Wqj04OrGcDihJimbu9+1JW0xp+eozmvdWti6Op6PD/cOd5XNqtPX+xx/tIVHibJqmtW8Jsa4r3+YRW2QhKJlZz/pxVSKG6LUoiKg5C7FzjkKgaAchsGGVjgG0Y+IQT6CYz4MPgUhFP9p7OCrma7klptCEN9/6/t//e7+YrYPN8v6Fi5JkpYY3/uQb85ODzZ11sglgw5T+5ZvvzH0T+z3nrPdBRTv7QwIhCigzA0AIgShaFaQARAujLgRVJcvGGeuMsczOGFzg69mxxLUxd70zKGgI09EpoobgAeF0Wnzre+8CuoBy5crO5z/x4sdvvuODXb9+Y2n7MqE5eOOdKxcufenn/u5wONy5thMkjMezr/3hn52Op3mWq4oxpqgrgOgbSGg8KoAIEVjL1rExzrAFRUak2H8tOkRjnY2RRkMOVGMBxJdna9Cn2nnf+tlsBgoRqhHgm9/70fVrV66fWyEKF69tF+lp9fa7e2/dGeW9jVvPPvvlV3V9WM/Gg2It6eUN2X//H//T/qNHde2zXi/M5knixrM5M0eLOgQPqoRk7IKOAEW65jda+7oQSCY6GbGKn7TC0PXyC7RSFV3oERTx49EIAA2RghJiHfC/fu1P//U//vkLS31VSXeWh9vn+jZbWltb2b0iw6wJrTha2ljX0H79z/7869/4y6zXG/R7PoixjIRN7dPMaghZmnYNPS7c/IW0JKKnBxFd3arKYnSCilGFQ5DgQ4jRSwjBB+9927YLf4KKsvQCIohKCEwq08p/5Wt/PKpLctb0UlpN5Fzm192c6qqtwVCSWOtsVVXf/f7baO2jk1E/s76pScRZV7QVIXoVssaxQSRAiq1j1xBgt/mioIAKqAAC0YsEiPQcW7hO9z3pAxaWGNNiUqST8VSFYgfMSIiKTFeevXk6mdV1bW3iennW67nEJXmaDzKLTXF6PJ/Mguh4MvYhFE3lrBEfesYQQlHXaZo2TW2cDRIHBbgIFKUz1kEWsyEABEJAJO992/oz0zzahnGpEoKKQPx3xKjOrWXxfjabhRCyNDFsGlBhzIf9z37xp4uyHT0+rIsCgQwbYgOo8/Hp7PBAmpqYo4VA1qbWei/EGIG7LqvOshINQTRacYALIb+AlkVjqaqR1aj13RNCANVupxGjIA+Lp6sbACYqZvPR+BRBnCVEDcTXb770j/7JP1vdOr97/WZQXxWTw6PDw8PD2XTctlPDkGa9JMmQOPhgEG/d/IQlLqtaRY1hJNQ4pYnUIx40mj8YqxHgCR11ugciI6HpqBrhzOSKoKkA+PTcCUBFWhFQ2NvbG49HS0uJl7bV8KWf/vLP/fwvFdXEpb3V5z5x7+Rukjoy1GpoRAIwIQZEBRUfxscn6MPnXnv9+Ojxo3u3DZvEceu9sZ1DFbwPwSN1slnlCZSrahzMdLwEBAqm0wtnk8BuJUCxE9BOTcRDEBFE3du710vsyvJKsOmv/PK/+sQLr7QNlQ+moIby3sXnPxUO7kawCCGEqq1NEYK2bTU5OXnjrbdv/tRnrj538zOPDv/owUPE1lo7GY+jh+mcJYR5Ua4SnXXjAKDYYU40tpg5hOiTgonRP1ENChIi3UpsBtiQCsRRFwAQ6IMHDzY2tzhJ/+Wv/tru5WeYjYLWTasAVcCtqy+czkdtNUNEEK2KUoK0wU8npx/f/vDg8PHn/sHP+X7/uedf/ODtt9//4F1SrDVEhWmdCyF43wJQN+TTpwcFTwAGsaMmEz2nWNiG2EvQzr+A6MiDEoIwGkQoZ/Mf/+id6XjyzPMv/+wv/sPzFy8RMiOG4INAdOEx33Dnr+Gjd6VR37bFbFbM51U9G50cPXpwd23z3O7zL909HG3t7l65devg5ACxxaDazRSpDd4SE0DoTEXtoB0Xw+V4Dgg+CMZhWRytAQARipduNhwNOYXQ+mI23X/0cO/e3cn4ZDye/vNf+dX1zfNJmhIRACqooJIhUQ1t+ItvfevW7gZwilS3dVvXoW3Lspoe7T+YluHTP/06p/2Ak+HK+pVnbzy48/7hw7tN64moqisFnU5nTdOoRPhDQqRFdiykZCRUBQCJPTEiGsNn/9QtQDEEGY+Ob9/+aDIeb25svfLpV/tLS20TEChJEwGIfkEQAUWXJE3byMz/6Ic/uLj++bWkjyChjhTYFPN5WzfDc7s7L7wGNqkb2dxcuvrsjb33flDPR7PyUepsVVVFWYpKURREBBoQgZ6aWD+Bk24ZCgpGVUMQ5O4CgiioArNhJna0le1sXbg8m87SNDOGg4hFcc4BIIGqYlSOCMACk9Hx7u7Vzc2Nk1l5bm3DV1MkFGmbpoS2NtZd/9yX8+WNw9F4dDq5umvO72xfvfmJ6Ww6mRYnk1lRlb5tZ217cngYggoAoQpFIUxPz/26gobOlQBV1SCIKKCgwGystcwc95iZl5aXRSTySpqlceNVFRQDoqg3bNI0PT79WCUsLy+JAA22/PFD347atg7SAtn1Z25duPlJZLp3d+/8uXOAIe33brz0Suvl7r0H1j42BKpYFtXo6NC4pJfnZTFrmgYXKuH/mxcQYlAlkSASRMV7r6LWujRNnUuNcdYmREYEnLOROeI02TBbY5xzSZrkeZa4xLDt5T1CCm27sboe2rB3XPDqpbL2bQgEmKxs9i6/+Ddvvd+IDJeW11bXQMFZu751cfXCLg03kK0PXhCm8/mD+3f37n/kRXu94XCwrIJRD515bGeIZKJxJ6pRuAFAHDk564gZkZgphABKUSc55zhWvTHW2iRxeZZkeZamiUudKpZ1Q0T9QU8Qpt61XlGBkHpr51rVajKdTudpmrrEsGFmA0iNIA/WTdrrdBrgbD6//cMfVlV9ejI2xl25eiXNUn1KKf/EOQhAEJEgIorIiUtFkA0b4kjRqMBIqChoHzwu/ur779+59/Du3Yf37hwcHUx8AOvS4cowyzNRRjJV0zpriQCZmqL0VSjU9s7vZv1elnHbhryXG8OEjIiqvvbCgwuXnn3BuNS3wIjVvCyn43Y2M+yGwz4inj+3RUwLNfoTsoIA4khIAACBsjwnIkJGNhKbCFVATZKsbfXf/Yev/vevfWM2a8tCjh/P3v3xncNHp0SYJG5paSmOcMqqclnPDdLm9OOiLnXlfP/ay6vbN9774KPNrS0VWVtZCSEQUQjiBUaTur+8+trnvnDh/HbrEZVUNXE2iB8uD4CQCLI0W19bJzwbJT+5WtD9MgJTELHWWmNUwVkXM00BnGVjuKra8WT8pc+/urI0sM4aCxubKxcubRGhMZwkibEmNqzsTGL08cd3l67dWHruU9S/OJ6W49H0/PnziXVJkkSRPy/KcVHu7R+n+WB7e/fm8y+2jUcy1rmTk9Hm5maWpbPZ3BorCoPBIEtSYn6y+YvZaexpgMiIaOslHwy9BDJGJU4FEVQQ6c79/cykm0tZllhnMcvss89eZkaXOCbK89QwqwKzBYGmlo0XXtu89blkuLq6eW7/0ZFNMiDq9XJCJYT5vHnv9sEf/sXbHx6eToOG/trNV18Hhtbg5vbFnWeup9mg9aEqa0Usq9Kg2dzcssZaY7r4QVXVqAghGeuIOHjx3rfeZ3muIsbabqKBBtC+f/v+1rnVjY0VZ13kyLIs+sOMkBDBJW4wGACoc84wDZaX8wuXeoPhbDafjMabGxuP9vf39vZevPViVVcfH4//+JtvTWdycDpp6vrOD9/5naODF65s5kubs8nJZ17/mesvfaqq27KplpcGjx4dOWsG6zkzO2ubVgFB4h09VRM9FmZDSGhYgtZN1ev12qbN0rSqawAEMmUV7ny8f+Hc8mCYW2ZkEJF5UWzhBqIigjE2SRNCyrPs6PDApmma98ti3suT0Gb7Dx4i08rSijX244PHf/ndd03e/9LLV0NVttX8/Xd//MM7+z++R5qup1Vx+foLLuvXTVM1dVG58ej08s7lNshsOh0OB0dHR0jxZpGIiiEkJUAyXqTX6xdl2cvT2Xg2XB6WZU1I8fbLw4PDx6PRpz7xTGqNgCIAMXrvRQWQo+u4ubHZtm2F9MGH7x0eH3328z/TgAeAwXDAvB1ANze3guDt+48f7J9+4fMvvfzcRjWr28Zn3F68cO6Pvv0D11/ryYnrD5oAKQAI7j98tL6+QoSPj45XV4cKqoASosJDADZELKKA4FtJ06ypayYu6zra6iEEJgbQ23fuI8Lm+jJiR9ls2TdSFEWSDuI9gO3t7f1HjwhNng13dnalFWCo6xJRk9Surq0nSdI0/qP7D3q9/uWtNQZNnE2sWVlZMtbmeTJIVkgHxOSDBpH5vECkPOs/Pj5ZXR0Qs6i2rY92/wKFkIg4BGG2dVPnedY0fjjsl1UdrWljjAI9PDzOU7uxurRwMNAwG8Pj0XQxQ0dmnM+LLM9W19fqtp3NZrPZzBg7n5dtG5xLgLFu6zt7e+ub64MsSdgZS0iapi5LnQR95pkr61sXSNkAKbIoicJkOhoMe8656azEJ+OmBQ+ogrMJqDJzMS+yPIv3O7z4aBcZY9ogR6eTYX8w6PXOsJiQDJu6rsUrdrY7G2t6/X6vnypWrS8n08lkMk3SZDwe97IeEY3n5bz057c2MmMUyLABRGsNgBbzcmd7d/Pirg+apMlsPi+KgonzXi9J3cnJlNjWbRM7XubO86KFbW6JsCgrVXJp2vrgrK2bVhWJKYgWc786HDjrABmQERABmBkAq3kd5/BJmmV5bqzJsh6qbdsGJMzn82iFDPt9JDgeTxhwe30lTUxs+gjJJUnjJc97K+vnlre2BcgkydHRYwVlY5IknU6rNMvmxQwUusJT4XhlQgUkbrNv8zwbTyZJkoQgbNi3HhGZqKjbWVGsLPUX8zLExcVkIprOisgq1tl409e61KWJAjBz07SHh48HvUGSOstmKbOvvXxtbdkqKIHGSRARHZ7Ml5fz/sD1lpaQzOloUlVVliaDQV9U0jSvm2Y+L6yz0eGKrSZFPx2RjLFENBwO57Micd1dxLyXqwgSHRweN63PU/vkttbiMcaURdU2ceADzhlEYsPWJc5aQqrrZj6db2xsWsMEkDJ+/tM3hjkaQ2zIMLEhYyhx4YuvPjfsm+HSCjEfHR2tra30enmSOmaez2aj07EEtZbTLKGnqoBM4ryoIUqMgyD9rFfOizzNULqJvqo+2n8M2uxsX3CGGcUQECqiIiozN6GdzGYCQIjOWBBh5HgxToL4ulldWcmcNYwStG6rNE3yNGHHgKoI1hhn+NLW8t/+7Ct5LzfWVlXbtL4/6KepHfbzpqrns3lVVCDBMGZprhCHqaIaaDabZ3mqKv1ejgjDpV5Z1kliQ/DetyICQPtHJ6DCBLSYqZ1dyEZEa+1sVkQDeHHRDQBARJg5z/O11dVoTvoQ6qZJkoTOUocIEFXVWZu41LfBJLaoq9WVNUbT6w2KommbUJXV4raJ5FnaDZ1AAICC9yqSpomxbK1xifVeEAgJq6pWUR/Cw8MTVAltrYvbdsYY52xcBjMXZV3XTRBJ00RE43VbUBHxrW/zXk6GBaCsawmBsPNYF5fXNISQZSkihOCZTNs0g8GAja2qZjSajEZTABoOlpIkQaV+rx9vSUTv8P8BBNYiUYQHH70AAAAASUVORK5CYII="
        width={2.125}
      />
      <span
        className="resultsDetailsBodyOrderedDoctor"
      >
        Dr.
        Reyes
      </span>
    </div>
    <span
      className="resultsDetailsBodyCollected"
    >
      Sat, Aug 14, 2021 - 10:17 AM
    </span>
    <span
      className="resultsDetailsBodyResulted"
    >
      Sat, Aug 14, 2021 - 10:17 AM
    </span>
    <span
      className="resultsDetailsBodyStatus"
    >
      Final
      <div
        className="resultsDetailsBodyStatusIndicator resultsDetailsBodyStatusIndicatorFinal"
      />
    </span>
  </div>
</div>
`;

exports[`ResultsDetails component should work with empty data 1`] = `
<div
  className="resultsDetails"
>
  <div
    className="resultsDetailsHeader"
  >
    <span
      className="resultsDetailsHeaderOrdered"
    >
      ordered by
    </span>
    <span
      className="resultsDetailsHeaderCollected"
    >
      collected on
    </span>
    <span
      className="resultsDetailsHeaderResulted"
    >
      resulted on
    </span>
    <span
      className="resultsDetailsHeaderStatus"
    >
      status
    </span>
  </div>
  <div
    className="resultsDetailsBody"
  >
    <div
      className="resultsDetailsBodyOrdered"
    >
      <span
        className="resultsDetailsBodyOrderedDoctor"
      />
    </div>
    <span
      className="resultsDetailsBodyCollected"
    />
    <span
      className="resultsDetailsBodyResulted"
    />
    <span
      className="resultsDetailsBodyStatus"
    >
      <div
        className="resultsDetailsBodyStatusIndicator resultsDetailsBodyStatusIndicatorundefined"
      />
    </span>
  </div>
</div>
`;
